### PR TITLE
copy channel purpose text implemented

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR adds the feature to copy channel purpose text.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24612

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on:   Android 12

#### Release Note

```release-note
Implemented copy channel purpose text functionality.
```

### CODE REVIEW

File: app/screens/channel_info/title/public_private/public_private.tsx [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet, Text, View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
 
+import SlideUpPanelItem, {ITEM_HEIGHT} from '@app/components/slide_up_panel_item';
+import {SNACK_BAR_TYPE} from '@app/constants/snack_bar';
+import {bottomSheet, dismissBottomSheet} from '@app/screens/navigation';
+import {bottomSheetSnapPoint} from '@app/utils/helpers';
+import {showSnackBar} from '@app/utils/snack_bar';
```
Comment (larkox) [inline app/screens/channel_info/title/public_private/public_private.tsx:15]: We import the files directly from the subfolder, not all the way up to the app. What does this means? Remove the `app/` from the imports. So...
`@app/components/...` => `@components/...`
`@app/constants/...` => `@constants/...`
etc...

File: app/screens/channel_info/title/public_private/public_private.tsx [hunk 2]
```diff
@@ -25,11 +34,62 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
...
+
+            bottomSheet({
+                closeButtonId: 'close-mardown-link',
+                renderContent,
+                snapPoints: [1, bottomSheetSnapPoint(2, ITEM_HEIGHT, bottom)],
+                title: intl.formatMessage({id: 'post.options.title', defaultMessage: 'Options'}),
+                theme,
+            });
+        }
+    }, [onCopy, managedConfig?.copyAndPasteProtection]);
```
Comment (larkox) [inline app/screens/channel_info/title/public_private/public_private.tsx:91]: This seems to depend also on the theme, bottom and intl.

File: app/screens/channel_info/title/public_private/public_private.tsx [hunk 3]
```diff
@@ -25,11 +34,62 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
...
     const theme = useTheme();
+    const managedConfig = useManagedConfig<ManagedConfig>();
+    const {bottom} = useSafeAreaInsets();
+
     const styles = getStyleSheet(theme);
     const publicPrivateTestId = 'channel_info.title.public_private';
 
+    const onCopy = useCallback(() => {
+        Clipboard.setString(purpose!);
+        dismissBottomSheet().then(() => {
```
Comment (larkox) [inline app/screens/channel_info/title/public_private/public_private.tsx:54]: We recently have seen that when we copy something into the clipboard, later Android version add an OS message, so this message becomes redundant.

Here is the PR that solves it in a different place. Feel free to apply this here, or take a look on how can it be done more centralized (maybe in `showSnackBar`? 0/5)

https://github.com/mattermost/mattermost-mobile/pull/7578/files

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you for the contribution! Just a few minor polishing comments.

File: app/screens/channel_info/title/public_private/public_private.tsx (hunk 1 above)
Comment (tanmaythole) [reply app/screens/channel_info/title/public_private/public_private.tsx]: Done 👍🏻

File: app/screens/channel_info/title/public_private/public_private.tsx (hunk 3 above)
Comment (tanmaythole) [reply app/screens/channel_info/title/public_private/public_private.tsx]: Done 👍🏻

File: app/screens/channel_info/title/public_private/public_private.tsx (hunk 2 above)
Comment (tanmaythole) [reply app/screens/channel_info/title/public_private/public_private.tsx]: Done 👍🏻

**Comment by tanmaythole** [comment]
Also,
Can you please add label as `hacktoberfest-accepted` in this PR or add topic as `hacktoberfest` in the repo, which helps to count my contribution in the ongoing hacktoberfest?

**Review by pvev — APPROVED** [review]
Thank you very much @tanmaythole for your help! 🎉  The code looks good to me. Good job! I left a couple of messages and a question, but nothing is a blocker. :shipit: 🚀

File: app/screens/channel_info/title/public_private/public_private.tsx [hunk 4]
```diff
@@ -25,11 +34,69 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
...
+    const intl = useIntl();
     const theme = useTheme();
+    const managedConfig = useManagedConfig<ManagedConfig>();
+    const {bottom} = useSafeAreaInsets();
+
     const styles = getStyleSheet(theme);
     const publicPrivateTestId = 'channel_info.title.public_private';
 
+    const onCopy = useCallback(async () => {
+        Clipboard.setString(purpose!);
```
[the remark this answers is not in the document]
Comment (larkox) [reply app/screens/channel_info/title/public_private/public_private.tsx:53]: I also thought about this, but this callback is only passed to a component that only exist if there is a purpose, so it should be safe to assume there will always be a purpose.

But maybe I am missing something.

[the remark this answers is not in the document]
Comment (tanmaythole) [reply app/screens/channel_info/title/public_private/public_private.tsx:53]: Yes, I agree on this, the callback is only passed to a component only if purpose exist.

**Review by larkox — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @tanmaythole - looks good to me 👍 

cc @larkox

**Pull request merged**